app_feup/lib/model/entities/restaurant.dart

```diff
@@ -0,0 +1,43 @@
+import 'dart:collection';
+import 'package:uni/model/utils/day_of_week.dart';
+
+import 'meal.dart';
+
+class Restaurant{
+  int id;
+  String name;
+  String reference; // Used only in html parser
+  HashMap<DayOfWeek, List<Meal>> meals;
+  Restaurant(String name, String reference, int id){
```

Comment:
As mentioned before, ideally this would be a constructor with just a few getters, but no means of changing data afterwards.